warehouse/forklift/legacy.py

```diff
@@ -1096,7 +1096,9 @@ def file_upload(request):
                 # --skip-existing functionality in twine
                 # ref: https://github.com/pypi/warehouse/issues/3482
                 # ref: https://github.com/pypa/twine/issues/332
-                "File already exists. See "
+                "File already exists "
+                + f"({filename!r}, with blake2_256 hash {file_hashes['blake2_256']!r})."
```

Comment:
The only concern I have here is whether `'blake2_256'` is guaranteed here, or should we do a `.get()` instead, so we don't raise a `KeyError`:

```suggestion
                + f"({filename!r}, with blake2_256 hash {file_hashes.get('blake2_256')!r})."
```

If the key is guaranteed to be in the dictionary, then this is fine as is!

Reply:
It'll be there as long as it's here, and I don't expect this to go anywhere anytime soon: https://github.com/pypi/warehouse/blob/b4b04243e1e77329edca590c27207aeff06a2ce2/warehouse/forklift/legacy.py#L1075